### ISSUE TITLE
Invoke-RestMethod on a bad SSL setup crash

### ISSUE THREAD

**Issue by AdrianArroyoCalle** [PowerShell/PowerShell#3588]
Steps to reproduce
------------------

Open Powershell. Execute:
```
Invoke-RestMethod https://192.168.0.173/feed
```
192.168.0.173 it's a local WordPress blog with SSL, though obviously configured only for my domain.

Expected behavior
-----------------

Ignore the SSL error or show an error. 

Actual behavior
---------------

PowerShell crashes without any clue of what happened and goes back to Bash.
![sslbug](https://cloud.githubusercontent.com/assets/3681517/25137080/fd443070-2456-11e7-819f-03c3a07ab416.png)

Environment data
----------------

Name                           Value                                           
----                           -----                                           
PSVersion                      6.0.0-alpha                                     
PSEdition                      Core                                            
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}                         
BuildVersion                   3.0.0.0                                         
GitCommitId                    v6.0.0-alpha.18                                 
CLRVersion                                                                     
WSManStackVersion              3.0                                             
PSRemotingProtocolVersion      2.3                                             
SerializationVersion           1.1.0.1                                         

```powershell
> $PSVersionTable

```

**Comment by SteveL-MSFT** [comment]
corefx uses libcurl internally on Linux.  can you try the same with curl?

**Comment by AdrianArroyoCalle** [comment]
```
curl: (51) SSL: no alternative certificate subject name matches target host name '192.168.0.173'. 
```
Tested in Bash and Powershell. It doesn't crash any of the shells unlinke Invoke-RestMethod.
![sslbug2](https://cloud.githubusercontent.com/assets/3681517/25147310/6c47d7ea-2477-11e7-8a78-b3f48ba2bb18.png)

**Comment by AdrianArroyoCalle** [comment]
Maybe this title is more appropiate for the issue: "Invoke-RestMethod on a bad SSL setup crash"

**Comment by SteveL-MSFT** [comment]
@AdrianArroyoCalle I can look into this if you can provide me instructions to setup a HTTPS endpoint on my machine that repos this

**Comment by AdrianArroyoCalle** [comment]
I'll try, but don't expect anything these days because I'm quite busy

**Comment by AdrianArroyoCalle** [comment]
Hi

Doing: 
```
Invoke-RestMethod https://imaginbank.com
```
also crashes PowerShell instead of giving an error. That is a public URL you can use to test.

**Comment by markekraus** [comment]
This looks like it was closed by #2006 an #4026. using the `-SkipCertificateCheck` option with certificates that have name mismatches works now.

@SteveL-MSFT Can we consider this one closed?

**Comment by SteveL-MSFT** [comment]
@markekraus `-SkipCertificateCheck` makes it work, but it shouldn't crash if not used.  I tried the repro on Windows and Linux (don't have Mac in my work office) and it returns an error as expected.  If someone finds it still crashes on Mac, please comment here.  Otherwise, I think this is closed.
